src/btree/bt_page.c

```diff
@@ -632,6 +632,18 @@ __inmem_row_leaf(WT_SESSION_IMPL *session, WT_PAGE *page)
             tombstone->prepare_state = WT_PREPARE_INPROGRESS;
             F_SET(tombstone, WT_UPDATE_PREPARE_RESTORED_FROM_DS);
             F_SET(upd, WT_UPDATE_RESTORED_FROM_DS);
+
+            /*
+             * If the update and tombstone both are from the same transaction, set the update also
+             * as prepare in-progress.
+             */
+            if (upd->txnid == tombstone->txnid && upd->durable_ts == tombstone->durable_ts &&
```

Comment:
How come we need to check timestamps here?

Reply:
Yes, we don't need to check the timestamps, transaction id is enough. Fixed with transaction id.